system/serial.h

```diff
@@ -19,7 +18,7 @@
 #define SERIAL_PORT_0x3E8   2
 #define SERIAL_PORT_0x2E8   3
 
-static const uint16_t serial_base_ports[] = { 0x3f8, 0x2f8, 0x3e8, 0x2e8 };
+static const uint16_t serial_io_ports[] = { 0x3f8, 0x2f8, 0x3e8, 0x2e8 };
 
 struct serial_port {
     bool enable;
```

Comment:
Moving enable and is_mmio last would avoid padding inside the struct. However, there may still be some padding between the struct and whatever lies after, which might even require an alignment of 8... so the code should probably be left that way, in the end :)